api/src/main/java/io/jsonwebtoken/SignatureAlgorithm.java

```diff
@@ -122,16 +122,27 @@
     private final boolean jdkStandard;
     private final int digestLength;
     private final int minKeyLength;
+    /**
+     * @deprecated This is just a workaround for https://bugs.openjdk.java.net/browse/JDK-8243551
+     */
+    @Deprecated
+    private final String oid;
 
     SignatureAlgorithm(String value, String description, String familyName, String jcaName, boolean jdkStandard,
                        int digestLength, int minKeyLength) {
+        this(value, description,familyName, jcaName, jdkStandard, digestLength, minKeyLength, null);
```

Comment:
Instead of passing `null`, is it possible to use the OIDs listed in the OpenJDK link I provided for the other algorithms?  Thoughts?

Reply:
Sorry, I took a while: https://github.com/jwtk/jjwt/pull/589#discussion_r436289039

Seems, that it's not necessary to have oids for every Signature-Algorithm. Should we have them anyways? Or could we leave them out, since the field is going to be removed anyways?

Reply:
Ah, I see!  Instead of `oid` then, we should have a `pkcs12Name` field (instead of `oid`) that captures the names represented in the table above.  This guarantees that everything is non-null as well as ensures a parallel variable (`jcaName` and `pkcs12Name` make more sense now).

Reply:
Yes, I agree. I renamed the field and made it a copy of jcaName for non-HmacSHA algorithms.